### PR TITLE
PROTON-2114:[CMake] Fixing an issue with cmake not compiling c++11 e…

### PR DESCRIPTION
**Pull request by ncau**
On versions prior to CMake 3.0 the C++11 examples are not being compiled as the CMake code does not correctly check for c++11 compatibility.

### CODE REVIEW

**Comment by gemmellr** [comment]
Can you please [raise and/or] reference a PROTON Jira key in your PR titles and commit logs.

https://issues.apache.org/jira/projects/PROTON

**Comment by ncau** [comment]
Updated the title and amended the commit message to contain the PROTON reference

File: cpp/examples/CMakeLists.txt [hunk 1]
```diff
@@ -56,6 +56,18 @@ if (DEFINED CMAKE_CXX_COMPILE_FEATURES)
   endif ()
   set(CMAKE_CXX_STANDARD ${STD})
   set(CMAKE_CXX_EXTENSIONS OFF)
+  else ()
```
Comment (astitcher) [inline cpp/examples/CMakeLists.txt]: This is indented wrongly

**Comment by astitcher** [comment]
Please also make sure that you have only a single squashed commit to rebase onto master.

**Comment by astitcher** [comment]
This PR is now obsolete

**Pull request closed without merging**